src/utils/reg.js

```diff
@@ -0,0 +1,52 @@
+
+function isIterator(val) {
+	return val && typeof val[Symbol.iterator] === 'function';
```

Comment:
This check will fail if `val` is an empty array or string, as they evaluate to false, while they are iterators... You should use `val != null` instead